File: 2997/solution.py
class Solution:
    def minOperations(self, nums: list[int], k: int) -> int:
        xor = 0
        for num in nums:
            xor ^= num
        xor ^= k
        res = 0
        while xor > 0:
            res += xor & 1
            xor >>= 1
        return res

        # time O(n)
        # space O(1)

File: 2997/solution.ts
function minOperations(nums: number[], k: number): number {
  let xor = nums.reduce((num, acc) => acc ^ num, 0);
  xor ^= k;
  let res = 0;
  while (xor) {
    if (xor & 1) {
      res++;
    }
    xor >>= 1;
  }
  return res;
}
